src/components/remote_control/src/commands/base_command_request.cc

```diff
@@ -0,0 +1,583 @@
+    SendResponse(
+        false, result_codes::kInvalidData, "Validation by schema failed");
+    return;
+  }
+  LOG4CXX_TRACE(logger_, "Request message validated successfully!");
+  using application_manager::HmiInterfaces;
+  if (!service_->IsInterfaceAvailable(HmiInterfaces::HMI_INTERFACE_RC)) {
+    LOG4CXX_WARN(logger_, "HMI interface RC is not available");
+    SendResponse(false,
+                 result_codes::kUnsupportedResource,
+                 "Remote control is not supported by system");
+    return;
+  }
+  LOG4CXX_TRACE(logger_, "RC interface is available!");
+  if (CheckPolicyPermissions() && CheckDriverConsent()) {
+    if (AqcuireResources()) {
+      Execute();  // run child's logic
+    }
+    // If resource is not aqcuired, AqcuireResources method will either
+    // send response to mobile or
+    // send additional request to HMI to ask driver consent
+  }
+}
+
+bool BaseCommandRequest::CheckPolicyPermissions() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!app_) {
+    LOG4CXX_ERROR(logger_, "Application doesn't registered!");
+    SendResponse(false, result_codes::kApplicationNotRegistered, "");
+    return false;
+  }
+
+  mobile_apis::Result::eType ret = service_->CheckPolicyPermissions(message_);
+  if (ret != mobile_apis::Result::eType::SUCCESS) {
+    LOG4CXX_WARN(logger_,
+                 "Function \"" << message_->function_name() << "\" (#"
+                               << message_->function_id()
+                               << ") not allowed by policy");
+
+    SendResponse(false, result_codes::kDisallowed, "");
+    return false;
+  }
+
+  return true;
+}
+
+application_manager::TypeAccess BaseCommandRequest::CheckModule(
+    const Json::Value& message) {
+  const std::string& module = ModuleType(message);
+  return service_->CheckModule(app_->app_id(), module)
+             ? application_manager::TypeAccess::kAllowed
+             : application_manager::TypeAccess::kDisallowed;
+}
+
+bool BaseCommandRequest::CheckDriverConsent() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  RCAppExtensionPtr extension = GetAppExtension(app_);
+  if (!extension) {
+    return false;
+  }
+  Json::Value value;
+  Json::Reader reader;
+  LOG4CXX_DEBUG(logger_, "Request: " << message_->json_message());
+  reader.parse(message_->json_message(), value);
+
+  application_manager::TypeAccess access = CheckModule(value);
+
+  if (IsAutoAllowed(access)) {
+    set_auto_allowed(true);
+    return true;
+  } else {
+    SendDisallowed(access);
+  }
+  return false;
+}
+
+bool BaseCommandRequest::AqcuireResources() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const Json::Value message_params =
+      MessageHelper::StringToValue(message_->json_message());
+
+  if (!IsResourceFree(ModuleType(message_params))) {
+    LOG4CXX_WARN(logger_, "Resource is busy.");
+    SendResponse(false, result_codes::kInUse, "");
+    return false;
+  }
+
+  AcquireResult::eType acquire_result = AcquireResource(message_params);
```

Comment:
Note there is another function where Acquire is spelled correctly

Reply:
@JackLivio issue fixed

Reply:
Confirmed Resolved 👍